apps/mobile-wallet/src/features/modals/modalSlice.ts

```diff
@@ -27,15 +27,27 @@ const modalSlice = createSlice({
   name: 'modals',
   initialState,
   reducers: {},
-  extraReducers(builder) {
+  extraReducers: (builder) => {
     builder
       .addCase(openModal, (state, action) => {
+        // Allow only one modal of the same type to be open at the same time
+        if (Object.values(state.entities).find((m) => m?.params.name === action.payload.name)) {
+          return
+        }
+
         modalAdapter.addOne(state, {
           id: Date.now(),
-          params: action.payload
+          params: action.payload,
+          isClosing: false
         })
       })
       .addCase(closeModal, (state, { payload: { id } }) => {
+        const modal = state.entities[id]
+        if (modal) {
+          modal.isClosing = true
+        }
```

Comment:
I think it's better to use the Redux provided functions for updating entries of adapters, in this particular case `updateOne` (similar to how you use `addOne` and `removeOne`), instead of mutating an object of the array like so. You can look for reference in the `addressesSlice`.

https://redux-toolkit.js.org/api/createEntityAdapter

Reply:
Done